pkg/controller/studyjob/katib_api_util.go

```diff
@@ -25,6 +25,25 @@ import (
 )
 
 func initializeStudy(instance *katibv1alpha1.StudyJob) error {
+	if validErr := validateStudy(instance); validErr != nil {
+		instance.Status.Condition = katibv1alpha1.ConditionFailed
+		return validErr
+	}
+
+	if getJobType(instance) != jobTypeNAS {
+		//Validate HP job
+		if validJobErr := validateHPJob(instance); validJobErr != nil {
+			instance.Status.Condition = katibv1alpha1.ConditionFailed
+			return validJobErr
+		}
+	} else {
+		//Validate NAS job
+		if validJobErr := validateNASJob(instance); validJobErr != nil {
+			instance.Status.Condition = katibv1alpha1.ConditionFailed
+			return validJobErr
+		}
+	}
+
```

Comment:
move validateHPJob and validateNASJob into validateStudy, and move validateStudy away from initializeStudy

Reply:
You want to put validateStudy function before initializeStudy?
Right here: https://github.com/kubeflow/katib/blob/master/pkg/controller/studyjob/studyjob_controller.go#L250 ?

Reply:
no, validateStudy has been moved to https://github.com/kubeflow/katib/blob/master/pkg/controller/studyjob/validating_webhook.go#L44